### ISSUE TITLE
blead cannot pass tests on AlmaLinux 8 (zlib linking problem)

### ISSUE THREAD

**Issue by toddr** [Perl/perl5#19685]
Right now when I build blead on AlmaLinux 8 (a Red Hat 8 clone), blead is failing to build.

My Configure options are:
```
$> BASE_DIR=/root/foo ./Configure -Dprefix=$BASE_DIR -Dusedevel -des -Doptimize='-Os' -Dusemymalloc='n' -Duseshrplib -Dinstallusrbinperl=no -Dscriptdir=$BASE_DIR/bin -Dscriptdirexp=$BASE_DIR/bin -Dman1dir=none -Dman3dir=none -Uuseithreads  
```

`make test` leads to:

```
#   Failed test 'ZLIB_VERSION (1.2.12) matches Compress::Zlib::zlib_version'
#   at t/cz-01version.t line 34.
#          got: '1.2.12'
#     expected: '1.2.11'
# 
# The version of zlib.h does not match the version of libz
# 
# You have zlib.h version 1.2.12
#      and libz   version 1.2.11
# 
# You probably have two versions of zlib installed on your system.
# Try removing the one you don't want to use and rebuild.
# Looks like you failed 1 test of 2.
../cpan/IO-Compress/t/cz-01version.t ............................... 
```

This is because for "reasons" on AlmaLinux 8, libperl links to `libnsl` which eventually links to libz. This then conflicts with the compiled version that `Compress::Raw::Zlib` uses. This test is revealing that Zlib.so is improperly using symbols from the shared library, not what was compiled statically into the so file.

This has been reported at https://github.com/pmqs/Compress-Raw-Zlib/issues/8 but given how close Perl 5.36 is to release, I thought I would report it here also.

**Comment by toddr** [comment]
This change was merged in b864a746559843cd8bc1720eaf14c83faeb8fcc7 which was part of v5.35.11

**Comment by neilb** [comment]
I see that @pmqs has released a new version of C:R:Zlib. If CPAN Testers is happy with that, I guess it can be merged into blead, or maybe @toddr you could grab the new release, drop it in and see if you can then build on AlmaLinux?

**Comment by toddr** [comment]
Yep. I can take a look.

**Comment by toddr** [comment]
Submitted this. https://github.com/Perl/perl5/pull/19725

**Comment by pmqs** [comment]
Sorry folks, there will be another release of C:R:Zlib. Another issue has been found.

**Comment by pmqs** [comment]
@toddr  New version of C:R:Zlib available -- https://github.com/pmqs/Compress-Raw-Zlib/releases/tag/v2.105

**Comment by rjbs** [comment]
I merged #19730
